buildconfig/CMake/PythonPackageTargetFunctions.cmake

```diff
@@ -33,18 +34,33 @@ CustomInstallLib = patch_setuptools_command('install_lib')
   endif ()
 
   set ( _egg_link_dir ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/${CMAKE_CFG_INTDIR} )
-  set ( _egg_link ${_egg_link_dir}/${pkg_name}.egg-link )
+
+  list(LENGTH ARGN _additional_args_length)
```

Comment:
We use the `cmake_parse_arguments` function in [other places](https://github.com/mantidproject/mantid/blob/master/buildconfig/CMake/SipQtTargetFunctions.cmake#L34) and it's quite neat for dealing with arguments. Can we use this?

Reply:
It seems a lot better - I'll have a look!